bundle-workflow/Pipfile

```diff
@@ -0,0 +1,12 @@
+[[source]]
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+pyyaml = "*"
```

Comment:
We should use a specific version, or at least a version range. "*" is going to be fragile.

Reply:
That's what Pipefile.lock is for. It locks at the current version that has been tested with. However, from the declaration point of view we think the future version of pyyaml will also work, until we know better. So there's no reason to specify a version here. It would be needed if we knew, for example, that version 3 works, but 4 breaks everything.

Reply:
Please don't assume I haven't read the documentation before commenting.

Breaking changes from major version updates is exactly why we should specify a major version here. If pyyaml ever releases version 4 (which, presumably, is a breaking change because of the major version bump) then the next time someone runs `pipenv update` it will pull in v4 and break everything. If we specify `3` then `pipenv update` will pull in `3.4` or whatever is the latest, but not attempt to pull in `4` and everything will continue to work.

Changing a dependency to a new major version should be a conscious decision, not a side-effect.

Reply:
> Please don't assume I haven't read the documentation before commenting.

Ask along if the docs aren't clear about how pipenv works.
 
> Breaking changes from major version updates is exactly why we should specify a major version here. If pyyaml ever releases version 4 (which, presumably, is a breaking change because of the major version bump) then the next time someone runs `pipenv update` it will pull in v4 and break everything. If we specify `3` then `pipenv update` will pull in `3.4` or whatever is the latest, but not attempt to pull in `4` and everything will continue to work.
 
> Changing a dependency to a new major version should be a conscious decision, not a side-effect.

Running `pipenv update` is that conscious decision and you have to test the outcome, then make a PR for the change to take effect. Nobody is running and making commits nilly willy of dependency changes. This is just by convention in every package manager, including pipenv: you put in these files the version as you know it, and let the tool figure out the best compatible matrix, write it into Pipfile.lock and then this becomes your current stable set.

Reply:
As discussed in person, we have a different view of what counts as a "conscious decision" about major version updates and how conservative our approach should be. Wildcards are fine; this is just a difference of opinion. No approach is inherently better than the other; there are just different tradeoffs.

> If I put a version in Pipfile, what do I need a Pipfile.lock for? That's not how the tool was designed.

The pipfile contains a version range (e.g. `>4`) and the Pipfile.lock contains the exact version that was resolved (e.g. `4.4.0`).